### DISCUSSION TITLE
Setting custom BIN colours - again

### DISCUSSION THREAD

**Discussion by SteveTWard** [robbrad/UKBinCollectionData#1077] [Q&A] [answered]
Apologies that this question has been asked and answered already recently, but the fix suggested doesn't work for me.
I modified the suggested fix as follows but I still get the invalid JSON format error. Please the contents of my file below
{
"FOOD BIN":
{
"icon": "mdi:trash-can",
"color": "red"
},
"GARDEN 240L":
{
"icon": "mdi:trash-can",
"color": "brown"
},
"REFUSE 180L":
{
"icon": "mdi:trash-can",
"color": "black"
},
"RECYCLING 240L":
{
"icon": "mdi:recycle",
"color": "green"
}
}

I assume that it is related to the names of the bins but I have tried all of the valid option I can see.
They are currently set to the value in the Bin type field on the Sensor for the specific Bin and I have also tried the Bin sensor name too.
My other question is the correct way to reference the json file i.e. should I just enter the file name or the full path in this format
/homeassistant/custom_components/uk_bin_collection/bincolors.json

**Comment by ABroodryk** [comment]
It looks like you just past the raw code into the into the section.

![image](https://github.com/user-attachments/assets/861e3e6f-d5d7-42cf-a95f-9fd7fe6c0dc5)

But I am yet to figure which names to use in the code for the bins... so one step closer...

**Reply by SteveTWard** [reply]
Ah I see thanks

**Reply by SteveTWard** [reply, marked as answer]
![image](https://github.com/user-attachments/assets/553f72c3-635a-4669-9953-9794ae93411f)
This now works for me using the bin type as the name

**Comment by pjuster** [comment]
I had the following bin types

![image](https://github.com/user-attachments/assets/04ab24bf-e572-4928-9209-6a45fc8b4736)
![image](https://github.com/user-attachments/assets/9f1b445a-fd52-4814-8859-0d27e66f50ff)

So I made this JSON using the bin type names
{
"Rubbish":
{
"icon": "mdi:trash-can",
"color": "Grey"
},
"Recycling":
{
"icon": "mdi:recycle",
"color": "Green/Brown"
}
}

pasted straight into

![image](https://github.com/user-attachments/assets/191ad27e-81e0-42d5-b47f-8be4675c7764)

and it worked (see above sensors showing the correct colours)
